src/pages/2019-04-22---bulding-a-slack-bot-to-help-handle-large-numbers-of-prs/index.md

```diff
@@ -0,0 +1,127 @@
+
+Clearly for Gatsby, we think PRs and the people writing the PRs are valuable — in the sense that our mission is to directly help build people and communities and to create tools that empower people and organizations. We want contributors because participating in Open Source directly helps their career and their contributions directly improves Gatsby which benefits everyone using it and the better Gatsby becomes, the more viable our commerical products & services become — which viability is what sustains and drives the whole process.
+
+## Managing Queues
+
+The next chapter is on “Managing Queues”. Really fun stuff.
+
+>In the last chapter, I showed you how to use an economic framework to guide your product development decisions. When we use such a framework, we often discover that cycle time has high economic leverage. If we can find cheap ways to reduce cycle time, our profits go up. But what causes long product development cycle time?
+
+He dives into the theory of queues (largely invented by a mathematician named Agner Karup Erlang (where have I heard that name before…🤔) at the Copehhangen Telephone Company who in 1909 was trying to determine how many phone lines were needed for a specific number of people) and how to manage queues — particularly when they get too long.
+
+The main principle from there we’ve taken to heart at Gatsby is principle 4: “**The Principle of High-Queue States: Most of the damage done by a queue is caused by high-queue states.**"
+
+“High-queue states” means in our context — too many PRs. PR count naturally flucuates as PRs come in semi-randomly and core maintainers have time to review and merge them. Sometimes these natural fluctuations mean the PR count soars to unusually high numbers. If not taken care of rapidly, this means the whole process will get bogged down.
+
+The rule we’ve adopted is that 120 PRs is a warning state when more people should shift over to working on PRs and 130 is red state when everyone should stop what they’re doing.
+
+This has been very successful as in the past, we’d at times get into 140+ PR count range and it’d kinda sit there for a few days until someone would finally heroically drive it down. Now PRs tend toward 100-115 state and we rarely move past that for long.
+
+## Multiple Classes of Queues?
+
+But it didn’t still feel like we were measuring things correctly. We’ve been inconsistent at responding to PRs and sometimes valuable PRs have slipped through the cracks and sat untouched for months. E.g. at the recent Drupalcon, an engineer came up to me after my talk and mentioned he’d opened a PR that hadn’t gotten merged. I looked and it was a great PR and luckily [Dustin Schau](<https://twitter.com/SchauDustin>) (also at the conferece) was able to review and merge it immediately. Luckily the developer was still in the middle of his Gatsby/Drupal project and hadn’t had to go into production with this bug unfixed.
```

Comment:
```suggestion
But it didn’t still feel like we were measuring things correctly. We’ve been inconsistent at responding to PRs and sometimes valuable PRs have slipped through the cracks and sat untouched for months. E.g. at the recent Drupalcon, an engineer came up to me after my talk and mentioned he’d opened a PR that hadn’t gotten merged. I looked and it was a great PR and luckily [Dustin Schau](https://twitter.com/SchauDustin) (also at the conferece) was able to review and merge it immediately. Luckily the developer was still in the middle of his Gatsby/Drupal project and hadn’t had to go into production with this bug unfixed.
```

☝️ is that actually a valid Markdown hyperlink? If it is -- TIL, and sorry for the suggestion 😆

Reply:
I didn't know this either! But [Typora](https://typora.io/), which I used to write the post, kept adding them so I kept it.